### ISSUE TITLE
Update denominator histograms

### ISSUE THREAD

**Issue by ktht** [HEP-KBFI/hh-multilepton#37]
So that the binning used in LO reweighting matches to the binning used in LO->NLO reweighting. However, this probably implies that we fork https://github.com/cms-hh/HHStatAnalysis (or Fabio's fork: https://github.com/fabio-mon/HHStatAnalysis/tree/NLO_model) under HEP-KBFI organization which I should be able to do. Once I've new denominators, then everyone needs to switch to our fork of the HHStatAnalysis repository.

**Comment by ktht** [comment]
As discussed in today's meeting, the change would imply more work while the benefits are questionable, since the only effect of lining up the binning in both LO and NLO reweighting would be that the event weights move between different bins of mHH when comparing NLO SM sample out-of-the-box to the sum of LO samples reweighted to NLO. This is a non-issue and hence does not need to be addressed.

**Comment by ktht** [comment]
Changes:
- added second set of denominator histograms with coarser binning so that it matches to the binning expected by the NLO reweighting procedure;
- the new histograms are now filled with generator-level weights instead of plain 1's;
- created V3 NLO HH reweighting method that uses denominator histograms for "differential" LO cross section (instead of some fit from theory), which implies no LO reweighting;
- disentangled NLO reweighting from LO reweighting, so that the V3 NLO method can be used without LO reweighitng;
- included bin volumes (product of bin widths) in V2 and V3 of NLO HH reweighting (V1 is not affected since the constant bin width factors out when normalizing the 1D histogram of differential cross sections to 1);
- made V3 reweighting method the default in all HH analyses;
- added test programs [`bin/testHHWeightInterfaceNLO.cc`](https://github.com/HEP-KBFI/tth-htt/blob/master/bin/testHHWeightInterfaceNLO.cc) and [`data/HHWeightInterfaceNLO/testHHWeightInterfaceNLO.py `](https://github.com/HEP-KBFI/tth-htt/blob/master/data/HHWeightInterfaceNLO/testHHWeightInterfaceNLO.py) in ttH repository that compute all possible reweighting weights from the same inputs, which allow us to compare V3 method implemented in C++ in our FW to the values that Fabio's Python interface produces.

Both test programs (that take 2017 HH->4V denominator histogram as input) produce the same values that are off by a constant factor of 1.001. So basically, the differential cross sections add up to the total theory cross section within 0.1% precision.

tldr: the NLO reweighting method works as expected, and is made the default in all HH analyses.

**Comment by ktht** [comment]
Forgot to update these lines:
https://github.com/HEP-KBFI/hh-multilepton/blob/42931681d687f5e14166a8ecacb0a6403a5f9385/src/HHGenKinematicsHistManager.cc#L56
https://github.com/HEP-KBFI/hh-multilepton/blob/42931681d687f5e14166a8ecacb0a6403a5f9385/src/DatacardHistManagerBase_hh.cc#L305
